### PR TITLE
[Docs]: Add advanced_guides/evaluation.md

### PR DESCRIPTION
**Pull request by soonera**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add advanced_guides/evaluation.md

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: docs/en/advanced_guides/evaluation.md [hunk 1]
```diff
@@ -1 +1,56 @@
...
+
+<!-- /TOC -->
+
+## Evaluation in MMEngine
+
+During model validation and testing, quantitative evaluation is often required. `Metric` and `Evaluator` have been implemented in MMEngine to perform this function. See [MMEngine Doc](https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/design/metric_and_evaluator.md).
+
+Model evaluation is divided into online evaluation and offline evaluation, as shown in the following figure:
+
+![img](https://user-images.githubusercontent.com/15977946/163718224-20a4970a-e540-4a3a-8b01-bf0a604c6841.jpg)
```
Comment (YuanLiuuuuuu) [inline docs/en/advanced_guides/evaluation.md]: I am a little bit confused by the diagram. Could you please give more detailed explanation?

File: docs/en/advanced_guides/evaluation.md [hunk 2]
```diff
@@ -1 +1,56 @@
...
+
+## Evaluation In MMSelfSup
+
+During pretrain, validation and testing are not included, so it is no need to use evaluation.
+
+During benchmark, evaluation is used on downstream tasks with other OpenMMLab projects like `MMDetection` and `MMSegmentation` .
```
Comment (YuanLiuuuuuu) [inline docs/en/advanced_guides/evaluation.md]:
```suggestion
During benchmark, the pre-trained model will use some other OpenMMLab repos, e.g. MMClassification and MMSegmentation, which have already implemented their own evaluation funtionalities. 
```

File: docs/en/advanced_guides/evaluation.md [hunk 3]
```diff
@@ -1 +1,56 @@
...
+
+Some evaluation functions are impleted in MMSelfSup, as follows:
```
Comment (YuanLiuuuuuu) [inline docs/en/advanced_guides/evaluation.md]:
```suggestion
But MMSelfSup also implements some custom evaluation functionalities, shown as below,
```

File: docs/en/advanced_guides/evaluation.md [hunk 4]
```diff
@@ -1 +1,114 @@
...
+predictions = load('prediction.pkl')
+
+results = evaluator.offline_evaluate(data, predictions, chunk_size=128)
+```
+
+## Evaluation In MMSelfSup
+
+During pretrain, validation and testing are not included, so it is no need to use evaluation.
+
+During benchmark, the pre-trained model will use some other OpenMMLab repos, e.g. MMClassification and MMSegmentation, which have already implemented their own evaluation functionalities.
```
Comment (fangyixiao18) [inline docs/en/advanced_guides/evaluation.md]:
```suggestion
During benchmark, the pre-trained models need other downstream tasks to evaluate the performance, e.g. classification, detection, segmentation, etc. It is recommended to run downstream tasks with other OpenMMLab repos, such as MMClassification or MMDetection, which have already implemented their own evaluation functionalities.
```

File: docs/en/advanced_guides/evaluation.md [hunk 5]
```diff
@@ -1 +1,114 @@
...
+
+But MMSelfSup also implements some custom evaluation functionalities, shown as below:
```
Comment (fangyixiao18) [inline docs/en/advanced_guides/evaluation.md]:
```suggestion
But MMSelfSup also implements some custom evaluation functionalities to support downstream tasks, shown as below:
```

File: docs/en/advanced_guides/evaluation.md [hunk 6]
```diff
@@ -1 +1,114 @@
...
+
+- [`knn_classifier()`](https://github.com/open-mmlab/mmselfsup/blob/dev-1.x/mmselfsup/evaluation/functional/knn_classifier.py)
```
Comment (fangyixiao18) [inline docs/en/advanced_guides/evaluation.md]:
```suggestion
- [`knn_classifier()`](mmselfsup.evaluation.functional.knn_classifier)
```

File: docs/en/advanced_guides/evaluation.md [hunk 7]
```diff
@@ -1 +1,114 @@
...
+It compute accuracy of knn classifier predictions, and is used in [KNN evaluation](https://github.com/open-mmlab/mmselfsup/blob/dev-1.x/tools/benchmarks/classification/knn_imagenet/test_knn.py#L179).
+
+```python
+...
+top1, top5 = knn_classifier(train_feats, train_labels, val_feats,
+                                        val_labels, k, args.temperature)
+...
+```
+
+- [`ResLayerExtraNorm`](https://github.com/open-mmlab/mmselfsup/blob/dev-1.x/mmselfsup/evaluation/functional/res_layer_extra_norm.py)
```
Comment (fangyixiao18) [inline docs/en/advanced_guides/evaluation.md]:
```suggestion
- [`ResLayerExtraNorm`](mmselfsup.evaluation.functional.ResLayerExtraNorm)
```

File: docs/en/advanced_guides/evaluation.md [hunk 8]
```diff
@@ -1 +1,114 @@
...
+    - [Online evaluation](#online-evaluation)
+    - [Offline evaluation](#offline-evaluation)
+  - [Evaluation In MMSelfSup](#evaluation-in-mmselfsup)
+  - [Customize Evaluation](#customize-evaluation)
+
+<!-- /TOC -->
+
+## Evaluation in MMEngine
+
+During model validation and testing, quantitative evaluation is often required. `Metric` and `Evaluator` have been implemented in MMEngine to perform this function. See [MMEngine Doc](https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/design/metric_and_evaluator.md).
```
Comment (fangyixiao18) [inline docs/en/advanced_guides/evaluation.md]:
```suggestion
During model validation and testing, quantitative evaluation is often required. `Metric` and `Evaluator` have been implemented in MMEngine to perform this function. See [MMEngine Doc](https://mmengine.readthedocs.io/en/latest/design/evaluation.html).
```

File: docs/en/advanced_guides/evaluation.md [hunk 9]
```diff
@@ -1 +1,114 @@
...
+        shared_head=dict(
+            type='ResLayerExtraNorm',
+            norm_cfg=norm_cfg,
+            norm_eval=False,
+            style='pytorch')))
+```
+
+## Customize Evaluation
+
+Custom `Metric` and `Evaluator`  are also supported, see [MMEngine Doc](https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/tutorials/evaluation.md)
```
Comment (fangyixiao18) [inline docs/en/advanced_guides/evaluation.md]:
```suggestion
Custom `Metric` and `Evaluator`  are also supported, see [MMEngine Doc](https://mmengine.readthedocs.io/en/latest/design/evaluation.html)
```

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

**Review by fangyixiao18 — APPROVED** [review]
